lib/pbench/server/cache_manager.py

```diff
@@ -892,19 +907,12 @@ def _discover_tarballs(self):
         resource_id of any tarballs we find in order to link them.
         """
         for file in self.path.iterdir():
-            if file.is_file() and Dataset.is_tarball(file):
-                tarball = Tarball(file, get_tarball_md5(file), self)
-                self.tarballs[tarball.name] = tarball
-                self.datasets[tarball.resource_id] = tarball
-                tarball.check_unpacked()
-            elif file.is_dir():
+            if file.is_dir():
                 md5 = file.name
                 for tar in file.iterdir():
-                    if tar.is_file() and Dataset.is_tarball(tar):
-                        tarball = Tarball(tar, md5, self)
-                        self.tarballs[tarball.name] = tarball
-                        self.datasets[tarball.resource_id] = tarball
-                        tarball.check_unpacked()
+                    self._add_if_tarball(tar, md5)
```

Comment:
Thanks for the refactor -- I think this code is much nicer now.  However, now I have a question.

We are iterating over the contents of the directory, but we don't really expect to find more than one tarball, right?  Is this really the most reasonable way to find it?  For instance, perhaps we should use a glob expression for `*.tar.xz` files, rather than trying them all and checking for that pattern?  Also, we might as well `break` out of the loop once we've found one, right?

Reply:
There should only be one in an isolation directory, but there might be multiples in the controller directory. I considered a `glob` in both cases, but I'm not sure there's really much advantage in the controller directory since we have to find all the isolator directories anyway, whereas in the isolator directories the pattern matching of the `glob` isn't clearly a win over the iteration on two files.